auslib/test/blobs/test_apprelease.py

```diff
@@ -1812,6 +1876,38 @@ def testCheckFailForUnsuportedAttributes(self):
 """)
         self.assertRaises(BlobValidationError, self.blobH3.validate, 'h', self.whitelistedDomains)
 
+    def testGetUrlGetsFromChannel(self):
+        updateQuery = {
+            "product": "h", "version": "30.0", "buildID": "10",
+            "buildTarget": "p", "locale": "l", "channel": "c1",
+            "osVersion": "a", "distribution": "a", "distVersion": "a",
+            "force": 0
+        }
+        patchKey = "partials"
+        patch = {"from": "h1"}
+        specialForceHosts = ["http://a.com"]
+
+        url = self.blobH2._getUrl(updateQuery, patchKey, patch, specialForceHosts)
+        expected_url = self.blobH2.get("fileUrls").get("c1").get("partials").get("h1")
+        self.assertEquals(expected_url, url)
+
+    def testGetUrlDoesntFallBackToAsterix(self):
+        updateQuery = {
+            "product": "h", "version": "30.0", "buildID": "10",
+            "buildTarget": "p", "locale": "l", "channel": "c1",
+            "osVersion": "a", "distribution": "a", "distVersion": "a",
+            "force": 0
+        }
+        patchKey = "partials"
+        patch = {"from": "h1"}
+        specialForceHosts = ["http://a.com"]
+
+        # overwrite the fileUrls section not to include the 'h1' product
+        self.blobH2['fileUrls']['c1']['partials'] = {'h2': 'http://a.com/h2-partial.mar'}
+
+        with self.assertRaises(ValueError):
+            self.blobH2._getUrl(updateQuery, patchKey, patch, specialForceHosts)
+
 
```

Comment:
@mozbhearsum I have been wondering if it would be better to test the `UnifiedFileUrlsMixin` instead having the same test for all the classes that make use of it......So just testing the `UnifiedFileUrlsMixin` as it's own independent unit(from where the term `unit tests` comes from)